fmriprep/utils/tests/test_derivative_cache.py

```diff
@@ -0,0 +1,41 @@
+import json
+from pathlib import Path
+
+import pytest
+
+from fmriprep.data import load as load_data
+from fmriprep.utils import bids
+
+
+@pytest.mark.parametrize('xfm', ['boldref2fmap', 'boldref2anat', 'hmc'])
+def test_transforms_found_as_str(tmp_path: Path, xfm: str):
+    spec = (
+        json.loads(load_data.readable('io_spec.json').read_text())
+        .get('queries')
+        .get('transforms')
+        .get(xfm)
+    )
```

Comment:
If we read from `io_spec.json`, then we reproduce bugs there here, instead of catching them. I would instead populate an example directory and look for what we expect.

```suggestion
@pytest.mark.parametrize('xfm', ['boldref2fmap', 'boldref2anat', 'hmc'])
def test_transforms_found_as_str(tmp_path: Path, xfm: str):
```